okhttp/src/main/java/okhttp3/internal/platform/AndroidSocketAdapter.kt

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.platform
+
+import okhttp3.Protocol
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+import java.nio.charset.StandardCharsets
+import javax.net.ssl.SSLSocket
+import javax.net.ssl.SSLSocketFactory
+import javax.net.ssl.X509TrustManager
+
+open class AndroidSocketAdapter internal constructor(private val packageName: String) {
+  internal lateinit var sslSocketClass: Class<in SSLSocket>
+  internal lateinit var paramClass: Class<*>
+  internal val setUseSessionTickets: Method by lazy {
+    sslSocketClass.getDeclaredMethod("setUseSessionTickets", Boolean::class.javaPrimitiveType)
+  }
+  internal val setHostname by lazy {
```

Comment:
Why are these lazy? I don’t really like lazy because you always pay for the lazy indirection, which will often make it more work than not lazy